service-runner.js

```diff
@@ -105,44 +100,103 @@ ServiceRunner.prototype._sanitizeConfig = function(conf, options) {
             // resolve relative paths
             configFile = path.resolve(self._basePath + '/' + configFile);
         }
-        return fs.readFileAsync(configFile)
-        .then(function(yamlSource) {
-            self.config = self._sanitizeConfig(yaml.safeLoad(yamlSource),
-                    self.options);
-
-            // Make sure we have a sane config object by pulling in
-            // package.json info if necessary
-            var config = self.config;
-            config.package = package_json;
-            if (config.info) {
-                // for backwards compat
-                var pack = config.package;
-                pack.name = config.info.name || pack.name;
-                pack.description = config.info.description || pack.description;
-                pack.version = config.version || pack.version;
-            }
-        })
-        .catch(function(e) {
-            console.error('Error while reading config file: ' + e);
-            process.exit(1);
-        });
+        action = fs.readFileAsync(configFile);
     }
+
+    var package_json = {};
+    try {
+        package_json = require(self._basePath + '/' + 'package.json');
+    } catch (e) {}
+
+    return action.then(function(yamlSource) {
+        self.config = self._sanitizeConfig(yaml.safeLoad(yamlSource), self.options);
+        // Make sure we have a sane config object by pulling in
+        // package.json info if necessary
+        var config = self.config;
+        config.package = package_json;
+        if (config.info) {
+            // for backwards compat
+            var pack = config.package;
+            pack.name = config.info.name || pack.name;
+            pack.description = config.info.description || pack.description;
+            pack.version = config.version || pack.version;
+        }
+    })
+    .catch(function(e) {
+        console.error('Error while reading config file: ' + e);
+        process.exit(1);
+    });
+};
+
+/**
+ * Updates the config and sets instance properties.
+ *
+ * @param conf a configuration.
+ *             If undefined - config is loaded from config.yaml file
+ *             If Object - treated as already parsed configuration
+ *             If sting - treated a serialized yaml config
+ */
+ServiceRunner.prototype.updateConfig = function updateConfig(conf) {
+    var self = this;
+
+    return self.loadConfig(conf)
+    .then(function() {
+        var config = self.config;
+        var name = config.package && config.package.name || 'service-runner';
+
+        // Set up the logger
+        if (!config.logging.name) {
+            config.logging.name = name;
+        }
+
+        // And the statsd client
+        if (!config.metrics.name) {
+            config.metrics.name = name;
+        }
+    });
+};
+
+ServiceRunner.prototype.processConfig = function() {
+    var self = this;
+    var config = self.config;
+
+    var name = config.package && config.package.name || 'service-runner';
```

Comment:
@d00rman we need this on line 188, and recalculating it is a bit more clear than use logger.name or metrics.name